core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -112,11 +112,11 @@ public void initialize(String name, Map<String, String> properties) {
     this.warehouseLocation = LocationUtil.stripTrailingSlash(inputWarehouseLocation);
     this.fs = Util.getFs(new Path(warehouseLocation), conf);
 
-    String fileIOImpl = properties.get(CatalogProperties.FILE_IO_IMPL);
-    this.fileIO =
-        fileIOImpl == null
-            ? new HadoopFileIO(conf)
-            : CatalogUtil.loadFileIO(fileIOImpl, properties, conf);
+    String fileIOImpl =
+        properties.getOrDefault(
+            CatalogProperties.FILE_IO_IMPL, "org.apache.iceberg.hadoop.HadoopFileIO");
+
+    this.fileIO = CatalogUtil.loadFileIO(fileIOImpl, properties, conf);
```

Comment:
Is it possible to add a test to validate that properties get set on the HadoopFileIO instance?

Reply:
@amogh-jahagirdar Could you please provide me a reference of tests for the Nessie catalog? I can plan a similar tests for the Hadoop catalog

Reply:
Sorry for the delay, it looks like Nessie Catalog does not really test that path (`TestNessieCatalog`) as far as I can tell but we don't necessarily need to follow the same here. I think we could add a test to `TestHadoopCatalog` which looks something like the following:

```
@ Test
public void testHadoopFileIOProperties() {
     String warehouseLocation = tableDir.getAbsolutePath();
    HadoopCatalog catalog = new HadoopCatalog();
    catalog.setConf(new Configuration());
    catalog.initialize(
        "hadoop", ImmutableMap.of(CatalogProperties.WAREHOUSE_LOCATION, warehouseLocation, "some-file-io-property", "some-file-io-value"));

    TableIdentifier tableIdent = TableIdentifier.of("db", "ns1", "ns2", "tbl");
    catalog.buildTable(tableIdent, SCHEMA).withPartitionSpec(SPEC).create();
    FileIO fileIO = catalog.newTableOps(tableIdent).io();

    assertThat(fileIO.properties().get(CatalogProperties.WAREHOUSE_LOCATION)).equals(warehouseLocation)
   assertThat(fileIO.properties().get("some-file-io-property)).equals("some-file-io.value")
}

```

Reply:
Thank you for your input @amogh-jahagirdar 
I have added a test for checking HadoopFileIO properties initialization. Also while adding tests, I noticed there is missing Null check for Hadoop config in `HadoopCatalog` so added that as well.
Please review the changes and let me know if any changes are required.

Reply:
I don't think the conf change is needed as @nastra pointed out it should always be set.